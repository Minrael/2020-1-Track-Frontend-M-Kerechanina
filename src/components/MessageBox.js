import React from 'react';

export default function messageToBox (messageText, key) {
    return (<p key = {key} className = 'message' dangerouslySetInnerHTML={{__html:messageText}}></p>)
}
