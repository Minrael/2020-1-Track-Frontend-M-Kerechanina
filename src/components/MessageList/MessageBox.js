import React from 'react';

const messageToBox = (messageText, key) => {
    return (<p key = {key} className = 'message' dangerouslySetInnerHTML={{__html:messageText}}></p>)
}

export default messageToBox;
