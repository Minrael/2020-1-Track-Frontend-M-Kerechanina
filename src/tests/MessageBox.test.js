import React from 'react'
import renderer from 'react-test-renderer'
import MessageBox from '../components/MessageList/MessageBox'

it('renders correctly', () => {
    const message = renderer
    .create(<MessageBox messageText = "What's up?" key = {1000} />)
    .toJSON();
    expect(message).toMatchSnapshot();
})


it('renders correctly with emoji', () => {
    const messageEmoji = renderer
    .create(<MessageBox messageText = '&#sad#&' key = {1111} />)
    .toJSON();
    expect(messageEmoji).toMatchSnapshot();
})