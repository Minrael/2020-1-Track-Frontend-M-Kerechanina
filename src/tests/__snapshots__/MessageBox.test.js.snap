// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="message"
  dangerouslySetInnerHTML={
    Object {
      "__html": Object {
        "messageText": "What's up?",
      },
    }
  }
/>
`;

exports[`renders correctly with emoji 1`] = `
<p
  className="message"
  dangerouslySetInnerHTML={
    Object {
      "__html": Object {
        "messageText": "&#sad#&",
      },
    }
  }
/>
`;
